tool/tsh/db.go

```diff
@@ -45,39 +46,67 @@ func onListDatabases(cf *CLIConf) error {
 	if cf.ListAll {
 		return trace.Wrap(listDatabasesAllClusters(cf))
 	}
-	tc, err := makeClient(cf, false)
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	var databases []types.Database
-	err = client.RetryWithRelogin(cf.Context, tc, func() error {
-		databases, err = tc.ListDatabases(cf.Context, nil /* custom filter */)
-		return trace.Wrap(err)
-	})
-	if err != nil {
-		return trace.Wrap(err)
-	}
 
-	proxy, err := tc.ConnectToProxy(cf.Context)
+	// Retrieve profile to be able to show which databases user is logged into.
+	profile, err := client.StatusCurrent(cf.HomePath, cf.Proxy, cf.IdentityFileIn)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	cluster, err := proxy.ConnectToCurrentCluster(cf.Context)
+	tc, err := makeClient(cf, false)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	defer cluster.Close()
 
-	// Retrieve profile to be able to show which databases user is logged into.
-	profile, err := client.StatusCurrent(cf.HomePath, cf.Proxy, cf.IdentityFileIn)
-	if err != nil {
-		return trace.Wrap(err)
-	}
+	var databases []types.Database
+	var roleSet services.RoleSet
+
+	err = client.RetryWithRelogin(cf.Context, tc, func() error {
+		errors := syncErrors{}
+		waitGroup := sync.WaitGroup{}
+		waitGroup.Add(2)
+
+		go func() {
+			defer waitGroup.Done()
+
+			clusterDatabases, err := tc.ListDatabases(cf.Context, nil /* custom filter */)
+			if err != nil {
+				errors.Add(trace.Wrap(err))
```

Comment:
I think that we can improve readability by using one of the  main GO principle: 
> Do not communicate by sharing memory; instead, share memory by communicating

What would you say about just inlining struct definition for result types like: 
```go
type databaseResult struct {
  databases []types.Database
  err error
}

type rolesResult struct {
  databases []types.Database
  err error
}
... 
if err != nil {
    dbResultC <- databaseResult{err: trace.Wrap(err))
    return
}
dbResultC <- databaseResult{databases: clusterDatabases)

``` 

and then using chan for communication:
```go

dbResults := <- databasesC
rolesResults  := <- rolesC

if err := trace.NewAgg(dbResult.err, rolesResults.err); err != nil {
   return err
}
```